x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java

```diff
@@ -64,8 +64,11 @@ public void runPolicy(String policy, IndexMetaData indexMetaData, ClusterState c
         }
         Step currentStep = getCurrentStep(stepRegistry, policy, indexSettings);
         if (currentStep == null) {
-            throw new IllegalStateException(
-                "current step for index [" + indexMetaData.getIndex().getName() + "] with policy [" + policy + "] is not recognized");
+            // This may happen in the case that there is invalid ilm-step index settings or the stepRegistry is out of
+            // sync with the current cluster state
+            logger.warn("current step [" + getCurrentStepKey(indexSettings) + "] for index [" + indexMetaData.getIndex().getName()
```

Comment:
Although I love that we want to provide the step info here, I have mixed feelings about the assertions that exist inside of `getCurrentStepKey`. I think this should be changed to throwing `IllegalStateException`. what do you think @colings86?

Reply:
I think the assertions are fine because ILM is in full control of the phase, action and step settings (they are INTERNAL settings so can't be touched by a user and will soon be moved to a custom index metadata object further locking down access to them. Therefore, I don't think its necessary to check that if one is set all three are set in production code but its useful to have the check in testing to make sure we don't do something silly so assertions feel right to me.